app/facades/food_facade.rb

```diff
@@ -0,0 +1,13 @@
+class FoodFacade
+  def self.foods_with_ingredient(ingredient)
+    json = FoodService.ingredient_query(ingredient)
+    json[:foods].map do |food|
+      Food.new(food)
+    end.first(10)
```

Comment:
You could look into pagination with your request so you don't need to limit it here -- a plus side of using pagination to limit the results you get back is that you won't need to iterate through ALL of the foods that are returned from the call before limiting it to 10.

Reply:
And going off of the comment/challenge I put on your controller above, if you could get this method to return both the total hits and the collection of food to the controller, that could also limit how many times you are calling the same endpoint.